dashboard/app/controllers/api/v1/pd/application/principal_approval_applications_controller.rb

```diff
@@ -7,5 +7,23 @@ def new_form
         application_guid: params.require(:application_guid)
       )
     end
+
+    def on_successful_create
+      # Approval application created, now score corresponding teacher application
+      teacher_application = Pd::Application::Teacher1819Application.find_by!(application_guid: @application.application_guid)
```

Comment:
After [this](https://github.com/code-dot-org/code-dot-org/pull/19486/files#diff-34764bcf50c0b88c777f2b2137fcb1bcR41), which I just merged, you can just do `@application.teacher_application`